fedimint-api/src/tiered_multi.rs

```diff
@@ -103,6 +104,41 @@ impl<T> TieredMulti<T> {
     pub fn get_mut(&mut self, amt: Amount) -> Option<&mut Vec<T>> {
         self.0.get_mut(&amt)
     }
+
+    /// Determines the denominations to use when representing an amount
+    ///
+    /// Algorithm tries to leave the user with a target number of `denomination_sets` starting
+    /// at the lowest denomination.  `self` gives the denominations that the user already has.
+    pub fn represent_amount<K>(
```

Comment:
Unclear to me is this for receiving or sending case.

Seems like there are generally two use cases:

* We are spending our spendable notes, and want to leave user with as many useful notes as possible.
* We are accepting payment (or making a change) and want to receive notes that will complement the existing ones to increase the chance of making next payments.

Reply:
That's right this is used in both cases, more generically any case where the user will receive ecash (either as change to self or as payment).

Reply:
```rust
/// Like above, but for cases when making change is not possible. Best effort.
fn calculate_outgoing_without_change(existing: Tiered<usize>, payment: Amount, min_destionation_tier_count: usize) -> Tiered<usize> /* payment */ { }
```

```rust
/// Like above but for receiving notes when payment is not possible. (Do we ever need it?)
fn calculate_receiving_without_change(existing: Tiered<usize>, payment: Amount, min_destionation_tier_count: usize) -> Tiered<usize> /* payment */ { }
```

Reply:
Choosing what notes to spend is the `select_coins` function.  We assume `existing` already has these notes removed.

There isn't a distinction necessary between notes received as a payment or as change.  In practice we are always doing one or the other, but even if we did both like in a PayJoin, we don't need to distinguish the outputs.